cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -154,33 +158,33 @@
    * @param program the name of the program
    * @param programStatuses the set of statuses to trigger the schedule. The schedule will be triggered if the status of
    *                        the specific program transitioned to one of these statuses.
-   * @return this {@link ScheduleBuilder}
+   * @return a {@link ScheduleCreationSpec}
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, String appVersion,
-                                                 ProgramType programType, String program,
-                                                 ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, String appVersion,
+                                              ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application version.
    *
-   * @see ScheduleBuilder#triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
-                                                 String program, ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
```

Comment:
Actually I feel this API doesn't make sense. What does it mean "the same application version" but namespace and application name are required to be specified? A different app in another namespace but the application version is the same as the current app? There's no meaningful use case.